ambry-router/src/main/java/com.github.ambry.router/PutOperation.java

```diff
@@ -810,13 +879,19 @@ private void prepareForSending() {
     /**
      * Do the actions required when the chunk has been completely built.
      * @param updateMetric whether chunk fill completion metrics should be updated.
+     * @param encryptContent {@code true} if chunk's content need to be encrypted. {@code false} otherwise
      */
-    void onFillComplete(boolean updateMetric) {
+    void onFillComplete(boolean updateMetric, boolean encryptContent) {
```

Comment:
Can the `encryptContent` variable be omitted? It seems like there is a value in `PutChunk` exposing an `isMetadataChunk()` method.